src/node_contextify.cc

```diff
@@ -1780,8 +1786,9 @@ static void CompileFunctionForCJSLoader(
         // This needs to call process.emit('warning') in JS which can throw if
         // the user listener throws. In that case, don't try to throw the syntax
         // error.
+        std::string warning_message = GetRequireEsmWarning(filename);
         should_throw =
-            ProcessEmitWarningSync(env, require_esm_warning).IsJust();
+            ProcessEmitWarningSync(env, warning_message.c_str()).IsJust();
```

Comment:
```suggestion
            ProcessEmitWarningSync(env, warning_message).IsJust();
```

That should work as well as `std::string` is convertible to `std::string_view`, and it's a best practice to use that conversion instead of `.c_str()` (because it's more performant and will not break if there are NUL bytes in the string – neither matters a lot here, but let's just stick to good practices across the codebase whenever we can)

Reply:
thank you